google/resource_kms_key_ring.go

```diff
@@ -0,0 +1,90 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudkms/v1"
+	"log"
+)
+
+func resourceKmsKeyRing() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKmsKeyRingCreate,
+		Read:   resourceKmsKeyRingRead,
+		Delete: resourceKmsKeyRingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func createKmsResourceParentString(project, location string) string {
```

Comment:
readability nit: I'm not a huge fan of `create` being used in functions that don't actually create something, especially in a code base like this where it realistically could be making API calls. How about just `kmsResourceParentString`?